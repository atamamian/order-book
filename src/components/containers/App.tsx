import { FC, ReactElement } from 'react';

const App: FC = (): ReactElement => (
  <div
    css={{ color: 'purple' }}
    data-testid="headline"
  >
    Order Book
  </div>
);

export default App;
