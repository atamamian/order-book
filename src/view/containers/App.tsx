import { FC, ReactElement } from 'react';
import { Global } from '@emotion/react';

import { globalStyles, wrapper } from '@/constants/styles';
import Header from './Header';
import FeedContainer from './FeedContainer';
import FeedFooter from './FeedFooter';

const App: FC = (): ReactElement => (
  <>
    <Global styles={globalStyles} />
    <div css={wrapper} data-testid="wrapper">
      <Header />
      <FeedContainer />
      <FeedFooter />
    </div>
  </>
);

export default App;
