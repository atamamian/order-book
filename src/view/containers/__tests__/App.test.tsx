import { fireEvent, render, waitFor } from '@testing-library/react';

import App from '../App';

describe('<App />', () => {
  it('renders', () => {
    const { queryByTestId } = render(<App />);
    expect(queryByTestId('wrapper')).not.toBeNull();
  });
  it('toggles feed', async () => {
    const { queryAllByTestId, queryByTestId } = render(<App />);
    fireEvent.click(queryByTestId('toggle-feed-button') as HTMLElement);
    await waitFor(() => expect(queryAllByTestId('spread-data')[0]?.innerHTML).toContain('ETH'));
  });
  it('kills feed', async () => {
    const { queryByTestId } = render(<App />);
    fireEvent.click(queryByTestId('kill-feed-button') as HTMLElement);
  });
  it('sets grouping when changed', () => {
    const { queryByTestId } = render(<App />);
    fireEvent.change(queryByTestId('market-group') as HTMLElement, {
      target: { value: 'Group 1' },
    });
    expect(queryByTestId('market-group')?.innerHTML).toContain('Group 1');
  });
});
