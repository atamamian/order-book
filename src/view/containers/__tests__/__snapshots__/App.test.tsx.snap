// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<body>
  <div>
    <div
      class="css-xbrsic-App"
      data-testid="wrapper"
    >
      <div
        class="css-14i7p5k-Header"
        data-testid="header"
      >
        <div
          data-testid="title"
        >
          Order Book
        </div>
        <div
          class="css-1ro6dqh-Spread"
          data-testid="spread-data"
        >
          Spread: 0.0 (0%)
        </div>
        <select
          class="css-1bn48zu-GroupSelect"
          data-testid="market-group"
          name="market-group"
        >
          <option
            value="0.5"
          >
            Group 0.5
          </option>
          <option
            value="1"
          >
            Group 1
          </option>
          <option
            value="2.5"
          >
            Group 2.5
          </option>
        </select>
      </div>
      <div
        class="css-qw88vv-FeedContainer"
      >
        <div
          class="css-1cl7lhk-FeedColumn"
        >
          <div
            class="css-t1a240-FeedRow"
          >
            <span>
              TOTAL
            </span>
            <span>
              SIZE
            </span>
            <span>
              PRICE
            </span>
          </div>
          <div
            class="css-1bkq1c3-FeedRow"
          >
            <span
              class="total"
            >
              0
            </span>
            <span
              class="size"
            >
              0
            </span>
            <span
              class="price"
            >
              0.00
            </span>
          </div>
          <div
            class="css-1bkq1c3-FeedRow"
          >
            <span
              class="total"
            >
              0
            </span>
            <span
              class="size"
            >
              0
            </span>
            <span
              class="price"
            >
              0.00
            </span>
          </div>
        </div>
        <div
          class="css-1lr1k6l-Spread"
          data-testid="spread-data"
        >
          Spread: 0.0 (0%)
        </div>
        <div
          class="css-1cl7lhk-FeedColumn"
        >
          <div
            class="css-q7h5nl-FeedRow"
          >
            <span>
              TOTAL
            </span>
            <span>
              SIZE
            </span>
            <span>
              PRICE
            </span>
          </div>
          <div
            class="css-1bt4gd7-FeedRow"
          >
            <span
              class="total"
            >
              0
            </span>
            <span
              class="size"
            >
              0
            </span>
            <span
              class="price"
            >
              0.00
            </span>
          </div>
          <div
            class="css-1bt4gd7-FeedRow"
          >
            <span
              class="total"
            >
              0
            </span>
            <span
              class="size"
            >
              0
            </span>
            <span
              class="price"
            >
              0.00
            </span>
          </div>
        </div>
      </div>
      <div
        class="css-4bbvyn-FeedFooter"
      >
        <div
          class="css-10sfhb4-ToggleFeedButton"
        >
          Toggle Feed
        </div>
        <div
          class="css-127tpaj-KillFeedButton"
        >
          Kill Feed
        </div>
      </div>
    </div>
  </div>
</body>
`;
